### PR TITLE
fix bug of switch and oom

### PR DESCRIPTION
**Pull request by zouxinyi0625**
1. fix switch bug in tryon.
2. fix oom problem by deleting sam model before training. Now, tryon is able to use ~8G.

### CODE REVIEW

File: scripts/easyphoto_tryon_infer.py [hunk 1]
```diff
@@ -236,6 +236,10 @@ def has_duplicate_uid(input_user_id, existing_user_ids):
         network_alpha = 64
         dataloader_num_workers = 16
 
+        ep_logger.info(f"Delete sam model before training to save CUDA memory.")
```
Comment (wuziheng) [inline scripts/easyphoto_tryon_infer.py:239]: 每次都得拉起~ 感觉有点不合适，但是先这样吧。

**Review by wuziheng — APPROVED** [review]
LGTM and test pass

**Comment by wuziheng** [comment]
1. GOOD:after delete sam, use only 5.6g GPU memory
2. GOOD: trans from tryon result 2 easyphoto 、t2i, i2i is great and seems fixed
3. BAD : When I test, only first transistion is success, it will always be the first image.

**Comment by zouxinyi0625** [comment]
> 1. GOOD:after delete sam, use only 5.6g GPU memory
> 2. GOOD: trans from tryon result 2 easyphoto 、t2i, i2i is great and seems fixed
> 3. BAD : When I test, only first transistion is success, it will always be the first image.

3. It is because the select function is put before the forward function. bug fixed.

**Pull request merged**